ports/esp32/network_wlan.c

```diff
@@ -452,7 +452,7 @@ STATIC mp_obj_t network_wlan_config(size_t n_args, const mp_obj_t *args, mp_map_
                         cfg.ap.channel = mp_obj_get_int(kwargs->table[i].value);
                         break;
                     }
-                    case MP_QSTR_dhcp_hostname: {
+                    case MP_QSTR_hostname: {
```

Comment:
I think we should provide some amount of backwards compatibility, at least for a few releases.  That's easy, just use:
```c
case MP_QSTR_hostname:
case MP_QSTR_dhcp_hostname : {
...
```